build_tools/aws-sdk-code-generator/spec/protocol_tests_spec_helper.rb

```diff
@@ -0,0 +1,291 @@
+# frozen_string_literal: true
+
+require 'rexml/document'
+require 'rspec/expectations'
+
+PROTOCOL_TESTS_IGNORE_LIST_PATH = File.join(File.dirname(__FILE__), 'protocol-tests-ignore-list.json')
+
+# This module contains helpers relating to protocol-tests
+module ProtocolTestsHelper
+  class << self
+
+    def check_ignore_list(protocol, test_id, test_type)
+      return nil if protocol.include?('extras')
+
+      ignore_list = JSON.parse(File.read(PROTOCOL_TESTS_IGNORE_LIST_PATH))[protocol]
+      ignore_list[test_type].find { |i| i.key?(test_id) }
+    end
+
+    # sets up paths for each protocol and its file paths
+    def fixtures
+      fixtures = {}
+      files = Dir.glob(File.join(File.dirname(__FILE__), 'protocol-tests', '**', '*.json'))
+      raise 'unable to locate spec files' if files.empty?
+
+      files.each do |path|
+        protocol = File.basename(path).split('.').first
+        input_output = path.split('/')[-2]
+        fixtures[protocol] ||= {}
+        fixtures[protocol][input_output] = path
+      end
+      fixtures
+    end
+
+    def each_test_case(context, fixture_path)
+      return unless fixture_path
+
+      Aws::Json.load_file(fixture_path).each do |suite|
+        context.context(suite['description']) do
+          suite['cases'].each do |test_case|
+            suite['metadata']['endpointPrefix'] ||= 'svc'
+            yield(self, suite, test_case, test_case['id'], test_case['description'])
+          end
+        end
+      end
+    end
+
+    # creates service client for req/resp testing
+    def client_for(suite, test_case, id)
+      name = "Service_#{id}"
+      unless Aws.const_defined?(name)
+        operation = test_case['given']
+        operation['http'] ||= { 'method' => 'POST', 'requestUri' => '/' }
```

Comment:
Is this correct as a default? Some protocols don't always POST to /

Reply:
I grandfathered this snippet from previous test runner so I'm not sure on the intention of setting that has default.  That being said -  all C2J translation of Smithy protocol tests always has these areas defined so I feel comfortable removing this fall back. 

(I will update the older "extras" tests to include this bit but there's  very few test cases)